cdc/log.cc

```diff
@@ -491,8 +493,11 @@ class transformer final {
 
     // TODO: is pre-image data based on query enough. We only have actual column data. Do we need
     // more details like tombstones/ttl? Probably not but keep in mind.
-    mutation transform(const mutation& m, const cql3::untyped_result_set* rs = nullptr) const {
+    std::optional<mutation> transform(const mutation& m, const cql3::untyped_result_set* rs = nullptr) const {
         auto ts = find_timestamp(*_schema, m);
+        if (ts == api::missing_timestamp) {
```

Comment:
Maybe this check could be done outside of `transform()` so preimage select is avoided when mutation is empty?

Reply:
Good idea - it not only improved performance in this case, but also simplified this patch.